Visual Studio Code materials and documentation

kdorheim opened the pull request:
**Describe development**

Add the materials and instructions for how to use VSC with Hector and update the instructions for VS, please see below for some questions/concerns. 

**Type of development** 

* MINOR: some functionality is added but backward compatible

**Checklist**
- [NA] [Roxygen](https://roxygen2.r-lib.org/) where applicable but also thorough and coherent inline documentation 
- [NA] [Unit tests](https://r-pkgs.org/tests.html) when applicable
- [x] Complies with the [Hector code style guide](https://jgcri.github.io/hector/articles/manual/StyleGuide.html)
- [x] The PR passes automated testing suite

kdorheim commented on .vscode/launch.json:
```diff
@@ -0,0 +1,57 @@
+{
```
@bpbond and @crvernon all of the project files/materials, the files associated with a platform/IDE for hector live in hector/project_files is it a problem that the `.vscode` doesn't live there too? Or does it not really matter because users can't see it since it is a`.dir`?

bpbond replied:
I don't think it matters? But don't know enough about VSCode to opine really.

kdorheim replied:
haha I mean I am by no means an expert, but it was a pretty good IDE to download and play around with after about 30 min I was able to build and run hector, much more initiative than Xcode and VS.

kdorheim commented on src/makefile.standalone:
```diff
@@ -1,7 +1,7 @@
-CXXFLAGS = -g $(INCLUDES) $(OPTFLAGS) $(CXXEXTRA) $(CXXPROF) $(WFLAGS) -MMD -std=c++11
+endif 
+CXXFLAGS = -g $(INCLUDES) $(OPTFLAGS) $(CXXEXTRA) $(CXXPROF) $(WFLAGS) -MMD -std=c++14 
 CFLAGS   = -g $(INCLUDES) $(OPTFLAGS) $(CCEXTRA) -MMD
```
@bpbond and @crvernon, when I was having issues building stand-alone hector on my machine BBL had me change this line when clang errors were causing `make hector` to fail. It is not clear if this is universally needed or only needed by a specific set up. It was pretty difficult to figure out... should it be included as a commented out line instead? Or should it just be a personal Hector setting?

bpbond replied:
1. I don't know.
2. I'd be inclined to has `c++14` be the default; this is over 6 years old now so it doesn't seem unreasonable to assume.

kdorheim commented on vignettes/manual/BuildHector.Rmd:
```diff
@@ -51,49 +55,17 @@ Hector's mandatory dependencies are listed in the `DESCRIPTION` file under `Impo
-
-## Linux
+## Command Line (Linux & Mac OSX)
 
```
Does the command line build/run work on windows?

bpbond replied:
Not that I know of.

kdorheim reviewed (commented):
@bpbond and @crvernon I've included my own review to point out specific issues I would like your input on

bpbond commented on vignettes/manual/BuildHector.Rmd:
```diff
@@ -156,6 +128,55 @@ sure to unset `CXXPROF`, clean the build directories with `make clean`,
+* *Search Paths-Header Search Paths*: "/usr/local/include /usr/local/lib/boost_1_52_0/" [or other header directories]
+* *Search Paths-Library Search Paths*: "/usr/local/lib/" [or other library directories]
+* *LLVM-Language-C++-C++ Standard Library*: libc++ (LLVM C++ standard library with C++11 support)
+* *Linking-Other Linker Flags*: "-lgsl -lgslcblas -lm"
```
We shouldn't need GSL flags?

kdorheim replied:
👍 make sense I suspect that some of these might be out of date

bpbond replied:
`boost_system` and `boost_filesystem` should be the only libraries we need to link against.

kdorheim replied:
I am not sure what this comment means, does that mean we can delete the following lines? 

```
* *Search Paths-Library Search Paths*: "/usr/local/lib/" [or other library directories]
* *LLVM-Language-C++-C++ Standard Library*: libc++ (LLVM C++ standard library with C++11 support)
* *Linking-Other Linker Flags*: "-lgsl -lgslcblas -lm"
```

bpbond replied:
We don't need the two gsl flags at the end . The others we need.

kdorheim commented:
@bpbond I believe that I addressed your comments with my commit from 4 days ago. Please let me know if you have any other suggestions, comments, or concerns. 

Thanks!

bpbond reviewed (approved):
Looks great @kdorheim — thanks!
